buildall

```diff
@@ -77,13 +77,16 @@ for DIST in $DISTS; do
    if [ "$repro_image_id" != "$built_image_id" ]; then
        log "$BASENAME:$DIST differs after a rebuild. Examine $built_image_id and $repro_image_id"
        log "to find the differences and fix the build to be reproducible again."
-       log "Running \`./dockerdiff $built_image_id $repro_image_id\` might be useful."
+       log "Changes (- first build, + second build):"
+       ./dockerdiff $built_image_id $repro_image_id || true
```

Comment:
Just a question out of curiosity: why do we allow this command to fail if we are exitting in the next line nonetheless?

Reply:
Good question.

The real reason is that I copied it from the second block where we are not directly
exiting.

The fake reason I'm going to give now to try and justify it is that it's not entirely
obvious that this command would cause the script to exit, so someone might
think that it would continue, and maybe even add another command after
this line that would never be called. "Explicit is better than implicit."

Reply:
Oh, and other context that might not be obvious is that `dockerdiff` returns a non-zero
exit code in normal operation, not just in errors.

The `diff` command returns 0 if there are no differences, and 1 if there are differences,
and `dockerdiff` replicates that. So in this case we expect there to be differences,
so we expect `dockerdiff` to return 1.

I wouldn't add `|| true` here to just catch an unexpected error exit from a command.
(and I don't think bash has a way to say "don't quit if this command returns 1, but
quit with an error if it's any other non-zero exit code.)

Reply:
I see, thank you very much for the explanation. I 100% agree on [Explicit is better than implicit](https://www.python.org/dev/peps/pep-0020/).

So as we already know the images differ, `dockerdiff` is going to return `1`. So the question then is if we are masking a real error on `dockerdiff` by using `|| true`, which I don't think it matters because we will see it in the output.

> (and I don't think bash has a way to say "don't quit if this command returns 1, but
quit with an error if it's any other non-zero exit code.")

I believe the only possible way to do that would be an `if` block.